jme3-vr/src/main/java/com/jme3/input/vr/lwjgl_openvr/LWJGLOpenVRInput.java

```diff
@@ -106,6 +163,87 @@ public LWJGLOpenVRInput(VREnvironment environment) {
         this.environment = environment;
     }
 
+    @Override
+    public void registerActionManifest(String actionManifestAbsolutePath, String startingActiveActionSets){
+        inputMode = InputMode.ACTION_BASED;
+        int errorCode = VRInput.VRInput_SetActionManifestPath(actionManifestAbsolutePath);
+
+        if ( errorCode != 0 )
+        {
+            logger.warning( "An error code of " + errorCode + " was reported while registering an action manifest" );
+        }
+        setActiveActionSet(startingActiveActionSets);
+    }
+
+    @Override
+    public void setActiveActionSet(String actionSet){
+        assert inputMode == InputMode.ACTION_BASED : "registerActionManifest must be called before attempting to fetch action states";
+
+
+        long actionSetHandle;
+        if (actionSetHandles.containsKey(actionSet)){
+            actionSetHandle = actionSetHandles.get(actionSet);
+        }else{
+            LongBuffer longBuffer = BufferUtils.createLongBuffer(1);
+            int errorCode = VRInput.VRInput_GetActionHandle(actionSet, longBuffer);
+            if ( errorCode != 0 )
+            {
+                logger.warning( "An error code of " + errorCode + " was reported while fetching an action set handle for " + actionSet );
+            }
+            actionSetHandle = longBuffer.get(0);
+            actionSetHandles.put(actionSet,actionSetHandle);
+        }
+
+        //Todo: this seems to imply that you could have multiple active action sets at once (Although I was not able to get that to work), allow multiple action sets
```

Comment:
It annoyed me that I couldn't get multiple action sets working. But then I didn't really understand why you'd want more than one action set, or why you'd want to turn them off. So I haven't worried about it too much

Reply:
The way multiple action sets could be used is, for example, if you want to have input per AppState. Or maybe one action set for interacting with the world and another for interacting with the user interface. There definitely are use cases, but I don't think it's a big deal if we don't support it, especially as the OpenVR documentation is a bit vague on how that is supposed to work.